project_5.html

```diff
@@ -0,0 +1,697 @@
+			</p>
+			<h3 class="mb-3" id="pj-heading-dfplayer">DFPLayer Mini MP3 Player</h3>
+			<p class="mb-3">
+				The Arduino Nano has too little memory to load and play audio files. Instead, we use a separate device to store and manage the audio media: The DFPlayer Mini is a small MP3 module that connects to an Arduino board (or other microcontrollers) over UART. 
+			</p>
+			<div class="mb-5">
+				<img class="img-fluid my-3 me-5 float-end" src="./assets/images/project_5/ops23_project5_dfplayermini.png" width = "150">
+				<p class="mb-3">
+					Though it supports a variety of control modes, we will use the module's serial mode. In serial mode, we need only connect the module's TX/RX, Power, Ground, and speaker pins. Using UART, the Arduino will send commands to the DFPlayer Mini to playback songs, pause, change volume, etc. This is facilitated by a custom Arduino library.
+				</p>
+				<p class="mb-3">
+					Explore more about the DFplayer Mini at its <a class="link-primary"
+					href="https://wiki.dfrobot.com/DFPlayer_Mini_SKU_DFR0299" target="_blank"
+					rel="noopener noreferrer">wiki page</a>.
+				</p>
+			</div>
+			<h4 class="mb-3 fw-light">Datasheet and Pinout Diagram</h4>
+			<p class="mb-3">
+				As an exercise of technical documentation literacy, we will not provide the pinout diagram. Instead, you can access the datasheet here [insert datasheet link after adapting to webpage]. 
+			</p>
+			<p class="mb-3">
+				Remember that you are looking for the following in the datasheet’s pinout diagram: <strong>TX</strong>, <strong>RX</strong>, <strong>Power</strong> (VCC), <strong>Ground</strong> (GND), <strong>Speaker +</strong>, and <strong>Speaker -</strong>. (Hint: Look at Section 2 of the datasheet).
+			</p>
+			<p class="mb-5">
+				Read Section 1.2, 1.4 for information about the supported file types, file naming conventions, and folder (a.k.a. directory) structure for the MicroSD card.
+			</p>
+			<h4 class="mb-3 fw-light">Using the DFPLayer Mini with Arduino</h4>
+			<p class="mb-3">
+				You can download the DFPlayer Mini library here [insert link here after adapting to webpage]. It can be used in your code with the following include:
+			</p>
+			<p class="mb-3">
+				<code class="pj-code language-arduino"> #include &lt;DFRobotDFPlayerMini.h&gt;</code>
+			</p>
+			<p class="mb-3">
+				Below is sample code that demonstrates the module in serial mode. The program plays a audio file named “0001.mp3” for five seconds, then pauses it for five seconds, then resumes. Use this as a <strong>reference for the commands to send to the MP3 module</strong>.
+			</p>
+			<p class="mb-3">
+				<span
+						class="text-blue">[Warning: Do not blindly copy-paste the sample code to your project deliverables. Read it, understand it, then adapt it to your own code.]</span>
+			</p>
+			<pre class="mb-5" style="white-space: pre-wrap;"><code class="pj-code language-arduino">#include &lt;SoftwareSerial.h&gt;
```

Comment:
This block is much larger than I expected. Let's place a height limit proportional to the viewport height and add a scrollbar on overflow. For reference, look at the code for the PDF viewer on the syllabus page.